sdk/src/main/java/com/amplitude/experiment/DefaultExperimentClient.kt

```diff
@@ -105,20 +111,41 @@ internal class DefaultExperimentClient internal constructor(
 
     override fun variant(key: String, fallback: Variant?): Variant {
         val variantAndSource = resolveVariantAndSource(key, fallback)
-        val variant = variantAndSource.variant
-        val source = variantAndSource.source
         if (config.automaticExposureTracking) {
-            exposureInternal(key, variant, source)
+            exposureInternal(key, variantAndSource)
         }
-        return variant
+        return variantAndSource.variant
     }
 
     override fun exposure(key: String) {
-        val variantAndSource = resolveVariantAndSource(key, null)
-        exposureInternal(key, variantAndSource.variant, variantAndSource.source)
+        val variantAndSource = resolveVariantAndSource(key)
+        exposureInternal(key, variantAndSource)
+    }
+
+    private fun exposureInternal(key: String, variantAndSource: VariantAndSource) {
+        legacyExposureInternal(key, variantAndSource.variant, variantAndSource.source)
+
+        // Do not track exposure for fallback variants that are not associated with a default variant.
+        val fallback = isFallback(variantAndSource.source)
+        if (fallback && !variantAndSource.hasDefaultVariant) {
+            return
+        }
+
+        val experimentKey = variantAndSource.variant.expKey
+        val metadata = variantAndSource.variant.metadata
+        var variant = if (!fallback && !variantAndSource.variant.isDefaultVariant()) {
+            variantAndSource.variant.key ?: variantAndSource.variant.value
+        } else {
+            null
+        }
+
+        val exposure = Exposure(key, variant, experimentKey, metadata)
+
+        userSessionExposureTracker?.track(exposure)
     }
 
-    private fun exposureInternal(key: String, variant: Variant, source: VariantSource) {
+
+    private fun legacyExposureInternal(key: String, variant: Variant, source: VariantSource) {
```

Comment:
In this function we can remove the userSessionExposureTracker calls, since we handle that in the non-legacy function.